src/components/TodoItem.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+export type Props = {
+  todo: Todo;
+  loading: boolean;
+  onToggle?: () => void;
+  onDelete?: () => void;
+  onUpdate?: (newTitle: string) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loading,
+  onDelete,
+  onToggle,
+  onUpdate,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const { title, completed, id } = todo;
+  const [isEditing, setIsEditing] = useState<boolean>(false);
```

Comment:
you don't need to provide the types for primitives
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```